File: J9Ch/src/J_8_Data_Structure/Required_10/_128_Hash_Function_Easy.java
package J_8_Data_Structure.Required_10;
import java.util.*;
/** 128 Hash Function
 * Easy

 * Created by tianhuizhu on 6/28/17.
 */
public class _128_Hash_Function_Easy {

    class Solution {
        public int hashCode(char[] key,int HASH_SIZE) {
            long ans = 0;
            for(int i = 0; i < key.length;i++) {
                ans = (ans * 33 + (int)(key[i])) % HASH_SIZE;
            }
            return (int)ans;
        }
    };
}

File: J9Ch/src/J_8_Data_Structure/Related_7/python_cpp/_128_Hash_Function_Easy.cpp
/** 128 Hash Function*/

class Solution {
public:
    int hashCode(string key,int HASH_SIZE) {
        int ans = 0;
        for(int i = 0; i < key.size();i++) {
            ans = (1LL * ans * 33 + key[i]) % HASH_SIZE;
        }
	return ans;
    }
};